operantanalysis/operantanalysis.py

```diff
@@ -1,108 +1,158 @@
-        allpokerpm += [pokerpm]
-        itipoke = 0
-        for x in range(itionidx, cueonidx):
-            if eventcode[x] == 1011 and timecode[x] >= (timecode[cueonidx] - cuelengthsec):
-                itipoke += 1
-        itipokerpm = itipoke / (cuelengthsec / 60)
-        allpokeitirpm += [itipokerpm]
-    
-    return round(statistics.mean(allpokerpm), 3), round(statistics.mean(allpokeitirpm), 3)
-
-
-def leverpressing(eventcode, lever1, lever2=False):
-    lever1presses = eventcode.count(lever1)
-    if bool(lever2):
-        lever2presses = eventcode.count(lever2)
+    cue_on = [i for i, event in enumerate(eventcode) if event == code_on]
+    cue_off = [i for i, event in enumerate(eventcode) if event == code_off]
+    iti_on = [i for i, event in enumerate(eventcode) if event == code_off or event == 'StartSession']
+    all_poke_rpm = []
+    all_poke_iti_rpm = []
+
+    for i in range(len(cue_on)):
+        cue_on_idx = cue_on[i]
+        cue_off_idx = cue_off[i]
+        iti_on_idx = iti_on[i]
+        cue_length_sec = (timecode[cue_off_idx] - timecode[cue_on_idx])
+        poke_rpm = ((eventcode[cue_on_idx:cue_off_idx].count('PokeOn1')) / (cue_length_sec / 60))
+        all_poke_rpm += [poke_rpm]
+        iti_poke = 0
+        for x in range(iti_on_idx, cue_on_idx):
+            if eventcode[x] == 'PokeOn1' and timecode[x] >= (timecode[cue_on_idx] - cue_length_sec):
+                iti_poke += 1
+        iti_poke_rpm = iti_poke / (cue_length_sec / 60)
+        all_poke_iti_rpm += [iti_poke_rpm]
+
+    return round(statistics.mean(all_poke_rpm), 3), round(statistics.mean(all_poke_iti_rpm), 3)
+
+
+def lever_pressing(eventcode, lever1, lever2=False):
+    """
+
+    :param eventcode: list of event codes from operant conditioning file
+    :param lever1: eventcode for lever pressing
+    :param lever2: optional parameter for second lever eventcode if two levers are used
+    :return: count of first lever presses, second lever presses, and total lever presses, as int
+    """
+    lever1_presses = eventcode.count(lever1)
+    if lever2:
+        lever2_presses = eventcode.count(lever2)
     else:
-        lever2presses = 0
-    totalleverpresses = lever1presses + lever2presses
+        lever2_presses = 0
+    total_lever_presses = lever1_presses + lever2_presses
 
-    return lever1presses, lever2presses, totalleverpresses
+    return lever1_presses, lever2_presses, total_lever_presses
+
+
+def lever_press_latency(timecode, eventcode, lever_on, lever_press):
+    """
+
+    :param timecode: list of times (in seconds) when events occurred
+    :param eventcode: list of events that happened in a session
+    :param leveron: event name for lever presentation
+    :param leverpress: event name for lever press
+    :return: the mean latency to press the lever in seconds
+    """
+    lever_on = [i for i, event in enumerate(eventcode) if event == lever_on or event == 'EndSession']
```

Comment:
Such selections are a really common pattern for you.  I would have come up with a helper function
`get_events_indices(eventcodes, eventtypes)` so here you would call
`level_on = get_events_indices(eventcodes, (lever_on, 'EndSession'))`.

Alternative, you could come up with a class `Events` to store your eventcodes, which would implement `.get_indices(eventtypes)`.

Reply:
I'll add this function in my new branch!